services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/PersistableFileDownload.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3;
+
+import java.time.Instant;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * An opaque token that holds the state and can be used to resume a
+ * paused download operation.
+ *
+ * @see S3TransferManager#downloadFile(DownloadFileRequest)
+ */
+@SdkPublicApi
+public final class PersistableFileDownload implements PersistableTransfer,
+                                                      ToCopyableBuilder<PersistableFileDownload.Builder, PersistableFileDownload> {
+    private final DownloadFileRequest downloadFileRequest;
+    private final long bytesTransferred;
+    private final Instant lastModified;
+
+    private PersistableFileDownload(DefaultBuilder builder) {
```

Comment:
Wondering if we should include a (perhaps optional) checksum of the bytes written so far, which a resume would validate before appending to an existing file.